### PR TITLE
Fix #2330 Added cancel option in filter dialog box.

### LINKED ISSUE #2330: No cancel option in filter dialog box in ScheduleFragment.java
**Actual Behaviour**

There is no cancel option on filter dialog box. The only way to cancel that dialog box is to press the filter text on that box, which creates a little confusion on how to cancel that dialog box. Also when the user presses back all the selected options are removed from the track but still the filter bar shows the selected filtering options.

**Expected Behaviour**

There should be a cancel option on the dialog box.

**Screenshots of the issue**

filter

**Would you like to work on the issue?**

Yes.

### PR DESCRIPTION
**Pull request by raj10071997**
Fixes #2330

Changes: Added the cancel option in the dialog box. Also removed the bug when the user presses back. When the user pressed back all the selected options were removed from the track but still the filter bar was showing the selected filtering options. Now on pressing back only the dialog box disappears.

Screenshot of the change:

changes

### CODE REVIEW

**Comment by dr0pdb** [comment]
@raj10071997 Screenshot please. 

> When the user pressed back all the selected options were removed from the track but still the filter bar was showing the selected filtering options

Please add this detail inside the issue also.

**Comment by raj10071997** [comment]
@srv-twry Added the screenshot of the change and also updated the issue. Sorry for the delay.

File: android/app/src/main/java/org/fossasia/openevent/core/schedule/ScheduleFragment.java [hunk 1]
```diff
@@ -233,8 +231,12 @@ public void filterSchedule() {
                         filterBar.setVisibility(View.VISIBLE);
                     } else {
                         filterBar.setVisibility(View.GONE);
+                        notifyUpdate(-1, selectedTracks);
                     }
-                });
+                })
+                .setNegativeButton("Cancel",((dialogInterface, i) -> {
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/core/schedule/ScheduleFragment.java]: Add proper spaces.

**Review by dr0pdb — CHANGES_REQUESTED** [review]

**Comment by raj10071997** [comment]
@iamareebjamal  Please review this PR.

**Comment by raj10071997** [comment]
@iamareebjamal Please merge this PR if no changes are required in it.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**